vck/src/commonMain/kotlin/at/asitplus/wallet/lib/agent/IssuerAgent.kt

```diff
@@ -209,85 +246,159 @@ class IssuerAgent(
+            null,
+            zlibService = zlibService,
+        )
+    }
+
+    private fun buildStatusListView(timePeriod: Int?): StatusListView {
+        return issuerCredentialStore.getStatusListView(
             timePeriod ?: timePeriodProvider.getCurrentTimePeriod(clock)
-        ).forEach { bitset[it] = true }
-        val input = bitset.toByteArray()
-        return zlibService.compress(input)?.encodeToString(Base64Strict)
+        )
     }
 
     /**
      * Revokes all verifiable credentials from [credentialsToRevoke] list that parse and validate.
      * It returns true if all revocations was successful.
      */
-    override fun revokeCredentials(credentialsToRevoke: List<String>): Boolean =
-        credentialsToRevoke.map { validator.verifyVcJws(it, null) }
-            .filterIsInstance<Verifier.VerifyCredentialResult.SuccessJwt>()
-            .all {
-                issuerCredentialStore.revoke(
-                    vcId = it.jws.vc.id,
-                    timePeriod = timePeriodProvider.getTimePeriodFor(it.jws.vc.issuanceDate)
-                )
-            }
+    override suspend fun revokeCredentials(credentialsToRevoke: List<String>): Boolean =
+        credentialsToRevoke.map {
+            validator.verifyVcJws(it, null)
+        }.filterIsInstance<Verifier.VerifyCredentialResult.SuccessJwt>().all {
+            issuerCredentialStore.setStatus(
+                vcId = it.jws.vc.id,
+                status = TokenStatus.Invalid,
+                timePeriod = timePeriodProvider.getTimePeriodFor(it.jws.vc.issuanceDate)
+            )
+        }
 
     /**
      * Revokes all verifiable credentials with ids from [credentialIdsToRevoke]
      * It returns true if all revocations was successful.
      */
     override fun revokeCredentialsWithId(credentialIdsToRevoke: Map<String, Instant>): Boolean =
         credentialIdsToRevoke.all {
-            issuerCredentialStore.revoke(
+            issuerCredentialStore.setStatus(
                 vcId = it.key,
-                timePeriod = timePeriodProvider.getTimePeriodFor(it.value)
+                status = TokenStatus.Invalid,
+                timePeriod = timePeriodProvider.getTimePeriodFor(it.value),
             )
         }
 
-    override fun compileCurrentRevocationLists(): List<String> {
+    override suspend fun provideStatusListToken(
+        acceptedContentTypes: List<StatusListTokenMediaType>,
+        time: Instant?,
+    ): Pair<StatusListTokenMediaType, Any> {
+        val preferedType = acceptedContentTypes.firstOrNull()
+            ?: throw IllegalArgumentException("Argument `acceptedContentTypes` must contain at least one item.")
+
+        return preferedType to when (preferedType) {
+            StatusListTokenMediaType.Jwt -> issueStatusListJwt(time)
+            StatusListTokenMediaType.Cwt -> issueStatusListCwt(time)
+        }
+    }
+
+    override suspend fun provideStatusListAggregation() = StatusListAggregation(
```

Comment:
That method is never used ... shouldn't there be a simple test for this?

Reply:
fixed in 93ce860696aa4077acb9a053a8b3eac9bee2733b